framework/test/bench/bundleinstall.cpp

```diff
@@ -45,6 +61,66 @@ class BundleInstallFixture
     framework.Stop();
     framework.WaitForStop(std::chrono::milliseconds::zero());
   }
+
+  void InstallConcurrently(benchmark::State& state, uint32_t numThreads)
+  {
+    using namespace std::chrono;
+
+    std::string bundleBasePath = "bundles\\test_bundle_";
+    auto framework = cppmicroservices::FrameworkFactory().NewFramework();
+    framework.Start();
+
+    // Generate paths to each bundle
+    std::vector<std::string> str5kBundles;
+    for (uint32_t count = 1; count <= 5000; count++) {
+      std::string copiedBundlePath = std::string(bundleBasePath);
+      copiedBundlePath.append(std::to_string(count));
+      str5kBundles.push_back(copiedBundlePath);
+    }
+
+    // Split up bundles per thread
+    std::vector<std::vector<std::string>> bundlesToInstallPerThread;
+    uint32_t currentIndex = 0;
+    for (uint32_t i = 0; i < numThreads; i++) {
+      std::vector<std::string> tempListOfBundles;
+
+      size_t numOfIterations =
+        (i == numThreads - 1 ? str5kBundles.size() / numThreads +
+                                 (str5kBundles.size() % numThreads)
+                             : str5kBundles.size() / numThreads);
+      for (uint32_t j = 0; j < uint32_t(numOfIterations); j++, currentIndex++) {
+        tempListOfBundles.push_back(str5kBundles[currentIndex]);
+      }
+
+      if (i == numThreads - 1) {
+        tempListOfBundles.push_back(str5kBundles[0]);
+      }
+
+      bundlesToInstallPerThread.push_back(tempListOfBundles);
+    }
```

Comment:
is there a STL algorithm that can do this? to avoid having to write loops.

Reply:
Unfortunately no. I rewrote it so that it is cleaner and uses the std::vector<T>(iterator, iterator) constructor.